task2/src/main/java/ru/cft/focusstart/Main.java

```diff
@@ -0,0 +1,69 @@
+package ru.cft.focusstart;
+
+import ru.cft.focusstart.InputOutput.UserFileReader;
+import ru.cft.focusstart.ShapesClasses.*;
+
+import java.io.FileNotFoundException;
+
+import static ru.cft.focusstart.InputOutput.UserFileReader.*;
+
+public class Main {
+
+    private static String INPUT_FILE;
+    private static String OUTPUT_FILE;
+
+    private static void argsParser(String[] args) {
+        try {
+            INPUT_FILE = args[0];
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Параметры командной строки не указаны! (Нет входящего файла)");
+            System.exit(0);
+        }
+        try {
+            OUTPUT_FILE = args[1];
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Файла для сохранения не указано. Выводим с консоль\n");
+        }
+    }
+
+    public static void main(String[] args) {
+
+        argsParser(args);//Парсим аргументы командной строки
+
+        try { //Проверяем, существует ли входящий файл
+            UserFileReader file = new UserFileReader(FILE_PATH + INPUT_FILE);
+            ShapeType shapeType = checkInputFileConfig(file); // Возвращаем ShapeType, если настройки верны
+
+            switch (shapeType) {
+                case CIRCLE:
+                    Shape circle = new Circle(getParsedParams());
+                    if (OUTPUT_FILE != null) {
+                        circle.write(FILE_PATH + OUTPUT_FILE);
+                    } else {
+                        circle.consoleOut();
+                    }
+                    break;
+                case TRIANGLE:
+                    Shape triangle = new Triangle(getParsedParams());
+                    if (OUTPUT_FILE != null) {
+                        triangle.write(FILE_PATH + OUTPUT_FILE);
+                    } else {
+                        triangle.consoleOut();
+                    }
+                    break;
+                case RECTANGLE:
+                    Shape rectangle = new Rectangle(getParsedParams());
+                    if (OUTPUT_FILE != null) {
+                        rectangle.write(FILE_PATH + OUTPUT_FILE);
+                    } else {
+                        rectangle.consoleOut();
+                    }
+            }
```

Comment:
Дублирование кода:

> if (OUTPUT_FILE != null) {
                        object.write(FILE_PATH + OUTPUT_FILE);
                    } else {
                        object.consoleOut();
                    }

Этого можно избежать т.к. методы write и consoleOut относятся к классу Shape 
```suggestion
           Shape shape = null;
            switch (shapeType) {
                case CIRCLE:
                    shape = new Circle(getParsedParams());
                    break;
                case TRIANGLE:
                    shape = new Triangle(getParsedParams());
                    break;
                case RECTANGLE:
                    shape = new Rectangle(getParsedParams());
                    break;
                default:
                    System.out.println("Неизвестный тип объект");
                    System.exit(0);
            }
            if (OUTPUT_FILE != null) {
                shape.write(FILE_PATH + OUTPUT_FILE);
            } else {
                shape.consoleOut();
            }
```